### PR TITLE
Combine weather set commands

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Combines weather commands into `Cmd_setfieldweather`:
```
Cmd_setrain
Cmd_setsunny
Cmd_sethail
Cmd_setsandstorm
BS_SetSnow
```

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
I need to look better bc I'm on mobile, but doesn't Snow and Hail also fail to be set if Primordial Sea is active?  If so, we need tests for those as well (+ Sandstorm for both primal weathers)

Besides that, looks good :)

**Comment by AlexOn1ine** [comment]
> I need to look better bc I'm on mobile, but doesn't Snow and Hail also fail to be set if Primordial Sea is active? If so, we need tests for those as well (+ Sandstorm for both primal weathers)
> 
> Besides that, looks good :)

Yeah it does. I didn't write a test because I thought it might be redundant since I already wrote one for `Desolate Land` but if you think they are important I can add them.

File: test/battle/weather/hail.c [hunk 1]
```diff
@@ -32,18 +32,20 @@ SINGLE_BATTLE_TEST("Hail damage does not affect Ice-type Pokémon")
 
 SINGLE_BATTLE_TEST("Hail fails if Desolate Land is active")
```
Comment (AsparagusEduardo) [inline test/battle/weather/hail.c:33]: Please update the name of the test to account for the change.

File: test/battle/weather/snow.c [hunk 2]
```diff
@@ -30,18 +30,20 @@ SINGLE_BATTLE_TEST("Snow multiplies the defense of Ice-types by 1.5x", s16 damag
 
 SINGLE_BATTLE_TEST("Snowscape fails if Desolate Land is active")
```
Comment (AsparagusEduardo) [inline test/battle/weather/snow.c:31]: Please update the name of the test to account for the change.

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

File: test/battle/weather/hail.c (hunk 1 above)
Comment (AlexOn1ine) [reply test/battle/weather/hail.c]: done

File: test/battle/weather/snow.c (hunk 2 above)
Comment (AlexOn1ine) [reply test/battle/weather/snow.c]: done

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**